src/components/SearchBar/index.jsx

```diff
@@ -2,13 +2,30 @@ import React from 'react';
 import { SearchField } from '@edx/paragon';
 import qs from 'query-string';
 import PropTypes from 'prop-types';
+import { getCurrentPage } from '../../components/SiteHeader';
 
+const SEARCH_PATH = '/search/';
 
 class SearchBar extends React.Component {
   constructor(props) {
     super(props);
     this.search = this.search.bind(this);
   }
+
+  getJournalIdForSearch() {
+    // If displaying the index page, search across all journals
+    // by using journalId=0
+    if (getCurrentPage(null, null, this.props.journalIndexId) === this.props.journalIndexId) {
```

Comment:
Took me a minute to grok what this was doing. Can you either add a comment right above saying "Check if we're on the index page" or pull the call into a variable and check the variable; i.e., 
```
const onIndexPage= getCurrentPage(...)
if (onIndexPage){
```
Clarity could also be added by renaming `getCurrentPage` to `getCurrentPageId`. But that's not a requirement.